File: test/output/code/csharp/ItemConfig.cs
// 自动生成于 sample.xlsx 时间: 2026-06-23T12:44:21.234Z
// 表: ItemConfig
// 请勿手动编辑此文件

using System.Collections.Generic;
using System.Text.Json;

namespace GameConfig
{
    /// <summary>
    /// ItemConfig 配置表接口
    /// </summary>
    public interface IItemConfig
    {
        /// <summary>道具ID</summary>
        int Id { get; set; }
        /// <summary>道具名称</summary>
        string Name { get; set; }
        /// <summary>类型</summary>
        string Type { get; set; }
        /// <summary>价格</summary>
        int Price { get; set; }
        /// <summary>可堆叠</summary>
        bool Stackable { get; set; }
        /// <summary>属性</summary>
        Dictionary<string, object> Attributes { get; set; }
    }

    /// <summary>
    /// ItemConfig 配置表
    /// </summary>
    public class ItemConfig : IItemConfig
    {
        /// <summary>道具ID</summary>
        public int Id { get; set; }
        /// <summary>道具名称</summary>
        public string Name { get; set; }
        /// <summary>类型</summary>
        public string Type { get; set; }
        /// <summary>价格</summary>
        public int Price { get; set; }
        /// <summary>可堆叠</summary>
        public bool Stackable { get; set; }
        /// <summary>属性</summary>
        public Dictionary<string, object> Attributes { get; set; }
    }

    /// <summary>
    /// ItemConfig 数据存储（key = id，可遍历）
    /// </summary>
    public static class ItemConfigData
    {
        /// <summary>全部数据，可通过 foreach / LINQ 遍历</summary>
        public static Dictionary<string, ItemConfig> All { get; } = new();

        /// <summary>根据 id 查询</summary>
        public static ItemConfig Get(string key)
        {
            return All.TryGetValue(key, out var value) ? value : null;
        }

        /// <summary>从 JSON 数据加载（自动兼容 verbose / compact / 纯数组格式）</summary>
        public static void ParseData(JsonElement obj)
        {
            var map = new Dictionary<string, ItemConfig>();
            JsonElement rows;

            if (obj.ValueKind == JsonValueKind.Array)
            {
                rows = obj;
            }
            else if (obj.TryGetProperty("data", out var dataProp) && dataProp.ValueKind == JsonValueKind.Array)
            {
                rows = dataProp;
            }
            else if (obj.TryGetProperty("d", out var dProp) && dProp.ValueKind == JsonValueKind.Array)
            {
                rows = dProp;
            }
            else
            {
                return;
            }

            var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
            foreach (var item in rows.EnumerateArray())
            {
                var record = JsonSerializer.Deserialize<ItemConfig>(item.GetRawText(), options);
                if (record != null)
                {
                    var keyValue = item.GetProperty("id").ToString();
                    map[keyValue] = record;
                }
            }
            All = map;
        }
    }
}

File: test/output/code/typescript/ItemConfig.ts
// 自动生成于 sample.xlsx 时间: 2026-06-23T12:21:54.097Z
// 表: ItemConfig
// 请勿手动编辑此文件

export interface IItemConfig {
  /** 道具ID */
  id?: number;
  /** 道具名称 */
  name?: string;
  /** 类型 */
  type?: string;
  /** 价格 */
  price?: number;
  /** 可堆叠 */
  stackable?: boolean;
  /** 属性 */
  attributes?: Record<string, unknown>;
}

export class ItemConfig implements IItemConfig {
  /** 道具ID */
  id?: number;
  /** 道具名称 */
  name?: string;
  /** 类型 */
  type?: string;
  /** 价格 */
  price?: number;
  /** 可堆叠 */
  stackable?: boolean;
  /** 属性 */
  attributes?: Record<string, unknown>;

  private static _dataMap: Record<string, IItemConfig> = {};

  /** 数据 Map，key = id，可直接 for...in / Object.keys 遍历 */
  public static get dataMap(): Record<string, IItemConfig> {
    return ItemConfig._dataMap;
  }

  /** 根据 id 查询 */
  public static Get(key: number | string): IItemConfig | undefined {
    return ItemConfig.dataMap[String(key)];
  }

  /** 从 JSON 数据加载（自动兼容 verbose / compact / 纯数组格式） */
  public static parseData(obj: any): void {
    // verbose 格式: { tableName, fields, data[] } → 取 data
    // compact 格式: { f, d[] } → 取 d
    // 纯数组格式: [...] → 直接使用
    let rows: any[];
    if (Array.isArray(obj)) {
      rows = obj;
    } else if (obj && Array.isArray(obj.data)) {
      rows = obj.data;
    } else if (obj && Array.isArray(obj.d)) {
      rows = obj.d;
    } else {
      // 已经是 key→value 映射
      ItemConfig._dataMap = obj;
      return;
    }
    const map: Record<string, IItemConfig> = {};
    for (const row of rows) {
      map[String(row.id)] = row as IItemConfig;
    }
    ItemConfig._dataMap = map;
  }
}
